cli/src/cmd/inspect.rs

```diff
@@ -0,0 +1,251 @@
+            ContractArtifactFields::Ir => f.write_str("ir"),
+            ContractArtifactFields::IrOptimized => f.write_str("irOptimized"),
+            // TODO: Add once these commands aren't broken at compiler-level
+            // ContractArtifactFields::Metadata => f.write_str("metadata"),
+            // ContractArtifactFields::UserDoc => f.write_str("userdoc"),
+            // ContractArtifactFields::Ewasm => f.write_str("ewasm"),
+        }
+    }
+}
+
+impl FromStr for ContractArtifactFields {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "abi" => Ok(ContractArtifactFields::Abi),
+            "bytecode" => Ok(ContractArtifactFields::Bytecode),
+            "deployedBytecode" | "deployed_bytecode" | "deployed-bytecode" | "deployed" |
+            "deployedbytecode" => Ok(ContractArtifactFields::DeployedBytecode),
+            "assembly" | "asm" => Ok(ContractArtifactFields::Assembly),
+            "asmOptimized" | "assemblyOptimized" | "assemblyoptimized" | "assembly_optimized" |
+            "asmo" => Ok(ContractArtifactFields::AssemblyOptimized),
+            "methodIdentifiers" | "method_identifiers" | "method-identifiers" | "mi" => {
+                Ok(ContractArtifactFields::MethodIdentifiers)
+            }
+            "gasEstimates" | "gas" | "gas_estimates" | "gas-estimates" | "gasestimates" => {
+                Ok(ContractArtifactFields::GasEstimates)
+            }
+            "storageLayout" | "storage_layout" | "storage-layout" | "storagelayout" | "storage" => {
+                Ok(ContractArtifactFields::StorageLayout)
+            }
+            "devdoc" => Ok(ContractArtifactFields::DevDoc),
+            "ir" => Ok(ContractArtifactFields::Ir),
+            "ir-optimized" | "irOptimized" | "iroptimized" | "iro" => {
+                Ok(ContractArtifactFields::IrOptimized)
+            }
+            // TODO: Add once these commands aren't broken at compiler-level
+            // "metadata" | "meta" => Ok(ContractArtifactFields::Metadata),
+            // "userdoc" => Ok(ContractArtifactFields::UserDoc),
+            // "ewasm" => Ok(ContractArtifactFields::Ewasm),
+            _ => Ok(ContractArtifactFields::Bytecode),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Parser)]
+pub struct InspectArgs {
+    #[clap(help = "the contract to inspect")]
+    pub contract: String,
+
+    #[clap(help = "the contract artifact field to inspect")]
+    pub mode: ContractArtifactFields,
+
+    /// All build arguments are supported
+    #[clap(flatten)]
+    build: build::BuildArgs,
+}
+
+impl Cmd for InspectArgs {
+    type Output = ();
+    fn run(self) -> eyre::Result<Self::Output> {
+        let InspectArgs { contract, mode, build } = self;
+
+        // Map mode to ContractOutputSelection
+        let mut cos = if let Some(v) = build.compiler.extra_output { v } else { vec![] };
+        if !cos.iter().any(|&i| i.to_string() == mode.to_string()) {
```

Comment:
can't we use standard `==` here?

Reply:
no because `i` is of type `ContractOutputSelection` whereas `mode` is the custom defined `ContractArtifactFields`. Maybe there's some comparison trait we could implement?